test/validation/api/atomic/atomic.c

```diff
@@ -706,6 +710,94 @@ static void test_atomic_non_relaxed_128_acq_rel(void)
 	}
 }
 
+static void test_atomic_bit_set_32(void)
+{
+	odp_barrier_wait(&global_mem->global_barrier);
+	odp_atomic_bit_set_u32(&global_mem->a32u_bit_set, 1U << (odp_thread_id() - 1));
```

Comment:
The thread ID could be taken before the barrier wait so that the atomic op is right after the barrier (as presumably the barriers are there to make it more likely that the atomic ops overlap).